conjur/init/init_controller.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+
+"""
+InitController module
+
+This module is the controller that facilitates all init actions
+required to successfully configure the conjurrc
+"""
+
+# Builtins
+import logging
+import os
+import sys
+
+# Third party
+from urllib.parse import urlparse
+
+# Internals
+import conjur.constants
+
+class InitController:
+    """
+    InitController
+
+    This class represents the Presentation Layer for the init command
+    """
+    force_overwrite = False
+    conjurrc_data = None
+    init_logic = None
+
+    def __init__(self, conjurrc_data, init_logic, force):
+        self.conjurrc_data = conjurrc_data
+        self.init_logic = init_logic
+        self.force_overwrite = force
+
+    def load(self):
+        """
+        Method that facilitates all method calls in this class
+        """
+        fetched_certificate = self.get_server_certificate()
+        self.write_certificate(fetched_certificate)
+
+        self.get_account_info(self.conjurrc_data)
+
+        self.write_conjurrc()
+
+        sys.stdout.write("Configuration initialized successfully!\n")
+        sys.stdout.write("To begin using the CLI, log in to the Conjur server by" \
+                         "running `conjur authn login`")
+
+    def get_server_certificate(self):
+        """
+        Method to get the certificate from the Conjur endpoint detailed by the user
+        """
+        # pylint: disable=line-too-long
+        if self.conjurrc_data.appliance_url is None:
+            self.conjurrc_data.appliance_url = input("Enter the URL of your Conjur server: ")
+
+        url = urlparse(self.conjurrc_data.appliance_url)
+        if url.scheme == "http":
+            return None
+
+        if self.conjurrc_data.cert_file is not None:
+            # Return None because we do not need to fetch the certificate
+            return None
+        # pylint: disable=logging-fstring-interpolation
+        logging.debug(f"Initiating a TLS connection with '{self.conjurrc_data.appliance_url}'")
+        fingerprint, fetched_certificate = self.init_logic.get_certificate(url.hostname, url.port)
+
+        sys.stdout.write(f"\nThe server's certificate SHA-1 fingerprint is:\n{fingerprint}\n")
+        sys.stdout.write("\nTo verify this certificate, it is recommended to run the following " \
+                         "command on the Conjur server:\n" \
+                         "openssl x509 -fingerprint -noout -in ~conjur/etc/ssl/conjur.pem\n\n")
+        trust_certificate = input("Trust this certificate? [no]: ")
+        if trust_certificate.lower() != 'yes':
+            raise ValueError("You decided not to trust the certificate")
+
+        return fetched_certificate
+
+    def get_account_info(self, conjurrc_data):
+        """
+        Method to fetch the account from the user
+        """
+        if conjurrc_data.account is None:
```

Comment:
negate and return instead of going into deeper indentation
```suggestion
        if conjurrc_data.account is not None:
            return
```